### PR TITLE
chore: Switch to gitleaks from git-secrets

### PR DESCRIPTION
**Pull request by sharat87**
The `gitleaks` project provides a more out-of-the-box experience and is able to find more than AWS keys in the files.

### CODE REVIEW

**Review by mohanarpit — APPROVED** [review]

File: contributions/CodeContributionsGuidelines.md [hunk 1]
```diff
@@ -25,16 +25,9 @@ We use [Github Flow](https://guides.github.com/introduction/flow/index.html), so
...
-   1. `brew install git-secrets` (macOS)
-   2. [Linux](https://github.com/awslabs/git-secrets#nix-linuxmacos)
-   3. [Windows](https://github.com/awslabs/git-secrets#windows)
-2. Register the secrets with git-secrets
-   1. `git secrets --register-aws`
-   2. This is very important step as it registers the regex to be scanned by the git-secrets.
-3. Verify the secrets are registered
-   1. `git secrets --list`
-4. Refer to [this repo](https://github.com/awslabs/git-secrets) for more information.
+1. Install `gitleaks`
```
Comment (KelvinOm) [inline contributions/CodeContributionsGuidelines.md:28]: Seems like [this docs](https://www.notion.so/appsmith/Getting-Started-e81ccc35463343b28e09c200cdbe5f42#bacce2d4aef0416094afac72b5715cd0) should be changed as well. Could you do this?

**Review by KelvinOm — APPROVED** [review]

File: app/client/.lintstagedrc.json [hunk 2]
```diff
@@ -7,5 +7,5 @@
   "cypress/**/*.json": ["prettier --write --cache"],
   "packages/**/*.{js,ts,tsx}": ["eslint --fix --cache"],
   "packages/**/*.{css,mdx,json}": ["prettier --write --cache"],
-  "*": ["git-secrets --scan --cached"]
+  "*": ["gitleaks protect --staged --verbose --no-banner"]
```
[the remark this answers is not in the document]
Comment (sharat87) [reply app/client/.lintstagedrc.json:10]: Yep. Am I missing something? 🤔

File: contributions/CodeContributionsGuidelines.md (hunk 1 above)
Comment (sharat87) [reply contributions/CodeContributionsGuidelines.md:28]: Thanks. There's actually nothing beyond `brew install gitleaks`. Should we just link to this Markdown document from that Notion page instead?

File: app/client/.lintstagedrc.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (KelvinOm) [reply app/client/.lintstagedrc.json:10]: I checked from my side, it looks like everything is working.
Снимок экрана 2023-11-27 в 09 57 40

File: contributions/CodeContributionsGuidelines.md (hunk 1 above)
Comment (KelvinOm) [reply contributions/CodeContributionsGuidelines.md:28]: I think the link should be enough.

**Review by KelvinOm — APPROVED** [review]

**Pull request merged**